src/shared/FeatureIntroControlContext/index.tsx

```diff
@@ -0,0 +1,145 @@
+import * as React from "react";
+
+/**
+ * These items are expected to be passed to the context provider
+ */
+interface FeatureIntroControlContextConfig {
+  id?: string;
+}
+
+/**
+ * Required: dismiss, heading, content
+ */
+interface FeatureIntroControl extends FeatureIntroControlContextConfig {
+  heading: React.ReactNode | undefined;
+  setHeading:
+    | React.Dispatch<React.SetStateAction<FeatureIntroControl["heading"]>>
+    | undefined;
+  content: React.ReactNode | undefined;
+  setContent:
+    | React.Dispatch<React.SetStateAction<FeatureIntroControl["content"]>>
+    | undefined;
+  image: React.ReactNode | undefined;
+  setImage:
+    | React.Dispatch<React.SetStateAction<FeatureIntroControl["image"]>>
+    | undefined;
```

Comment:
Oh it looks like the context is either the interface or undefined, so I think we can require the setters here
```suggestion
  setImage: React.Dispatch<React.SetStateAction<FeatureIntroControl["image"]>>;
```

Reply:
If these elements are called outside of the Context they will be undefined and we want to return the raw element, but every child calls `useFeatureIntroControlInternalContext` to see if they are defined before returning themselves, I think thats why I had these as undefined.

Reply:
ahh yep that totally makes sense, how would you feel about `useFeatureIntroControlInternalContext` returning undefined when not in a context provider, which would change usage a bit to something like
```
const featureContext = useFeatureIntroControlInternalContext()

const id = featureContext?.id
... make the element
return featureContext ? null : element
```

I don't have strong feelings on that, so I'll definitely leave it up to you :)